cwms-data-api/src/main/java/cwms/cda/data/dto/locationlevel/VirtualLocationLevel.java

```diff
@@ -0,0 +1,352 @@
+			this.specifiedLevelId = specifiedLevelId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withParameterTypeId(String parameterTypeId) {
+			this.parameterTypeId = parameterTypeId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withParameterId(String parameterId) {
+			this.parameterId = parameterId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withLevelUnitsId(String levelUnitsId) {
+			this.levelUnitsId = levelUnitsId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withLevelDate(ZonedDateTime levelDate) {
+			this.levelDate = levelDate;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withLevelComment(String levelComment) {
+			this.levelComment = levelComment;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withDurationId(String durationId) {
+			this.durationId = durationId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeValue(BigDecimal attributeValue) {
+			this.attributeValue = attributeValue;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeUnitsId(String attributeUnitsId) {
+			this.attributeUnitsId = attributeUnitsId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeParameterTypeId(String attributeParameterTypeId) {
+			this.attributeParameterTypeId = attributeParameterTypeId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeParameterId(String attributeParameterId) {
+			this.attributeParameterId = attributeParameterId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeDurationId(String attributeDurationId) {
+			this.attributeDurationId = attributeDurationId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withAttributeComment(String attributeComment) {
+			this.attributeComment = attributeComment;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withLocationLevelId(String locationId) {
+			this.locationId = locationId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withOfficeId(String officeId) {
+			this.officeId = officeId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withInterpolateString(String interpolateString) {
+			this.interpolateString = interpolateString;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel build()
+		{
+			return new VirtualLocationLevel(this);
+		}
+	}
+
+	@JsonPOJOBuilder
+	@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+	@JsonDeserialize(builder = Constituent.Builder.class)
+	@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class, aliases = {Formats.DEFAULT, Formats.JSON})
+	public static final class Constituent
```

Comment:
There's the rating and level types of constituents, there should be two subclasses